monitoring/resourse_lakehouse_monitor.go

```diff
@@ -0,0 +1,139 @@
+	return tmpSchema
+}
+
+type Snapshot struct {
+}
+type Monitor struct {
+	AssetsDir                string                                   `json:"assets_dir"`
+	BaselineTableName        string                                   `json:"baseline_table_name,omitempty"`
+	CustomMetrics            []catalog.MonitorCustomMetric            `json:"custom_metrics,omitempty"`
+	DataClassificationConfig *catalog.MonitorDataClassificationConfig `json:"data_classification_config,omitempty"`
+	FullName                 string                                   `json:"table_name"`
+	InferenceLog             *catalog.MonitorInferenceLogProfileType  `json:"inference_log,omitempty"`
+	Notifications            []catalog.MonitorNotificationsConfig     `json:"notifications,omitempty"`
+	OutputSchemaName         string                                   `json:"output_schema_name"`
+	Schedule                 *catalog.MonitorCronSchedule             `json:"schedule,omitempty"`
+	Snapshot                 Snapshot                                 `json:"snapshot,omitempty"`
+	SkipBuiltinDashboard     bool                                     `json:"skip_builtin_dashboard,omitempty"`
+	SlicingExprs             []string                                 `json:"slicing_exprs,omitempty"`
+	TimeSeries               *catalog.MonitorTimeSeriesProfileType    `json:"time_series,omitempty"`
+	WarehouseId              string                                   `json:"warehouse_id,omitempty"`
+}
+
+type MonitorInfo struct {
+	Monitor
+	DriftMetricsTableName   string                    `json:"drift_metrics_table_name,omitempty"`
+	ProfileMetricsTableName string                    `json:"profile_metrics_table_name,omitempty"`
+	Status                  catalog.MonitorInfoStatus `json:"status,omitempty"`
+	TableName               string                    `json:"table_name,omitempty"`
+}
+
+func ResourceLakehouseMonitor() common.Resource {
+	monitorSchema := common.StructToSchema(
+		Monitor{},
+		func(m map[string]*schema.Schema) map[string]*schema.Schema {
+			common.MustSchemaPath(m, "output_schema_name").Required = true
+
+			common.MustSchemaPath(m, "inference_log", "granularities").Required = true
+			common.MustSchemaPath(m, "inference_log", "granularities").Optional = false
+
+			common.MustSchemaPath(m, "inference_log", "problem_type").Required = true
+			common.MustSchemaPath(m, "inference_log", "problem_type").Optional = false
+
+			common.MustSchemaPath(m, "inference_log", "timestamp_col").Required = true
+			common.MustSchemaPath(m, "inference_log", "timestamp_col").Optional = false
+
+			common.MustSchemaPath(m, "inference_log", "prediction_col").Default = "prediction"
+
+			common.MustSchemaPath(m, "time_series", "granularities").Required = true
+			common.MustSchemaPath(m, "time_series", "granularities").Optional = false
+
+			common.MustSchemaPath(m, "time_series", "timestamp_col").Required = true
+			common.MustSchemaPath(m, "time_series", "timestamp_col").Optional = false
```

Comment:
These customizations aren't reflected in the Terraform schema because the `Schema` field is set to `readSchema`